Add the ability for the scrollbox to scroll in both direction

bbazukun123 opened the pull request:
Added `bidirectionalScroll` to the `ScrollBoxOptions`, which allows the scroll-box to be scrolled both vertically and horizontally to accommodate overflows in both directions.

CyberDex reviewed (changes requested):
Please add a `bidirectional` option to a
`export type ListType = 'horizontal' | 'vertical';`
of [List.ts](https://github.com/pixijs/ui/blob/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/List.ts#L309)

This way it will be consistent and will not require additional setting for the `Scrollbox`, together with that it will improve the `type` setting of the `List` as now it is not cleat enough. 

It will just work out of the box for `List` as this is essentially how it works when `type` is undefined. So you will just have to add 
```
case 'bidirectional':
default:
````
[over here](https://github.com/pixijs/ui/blob/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/List.ts#L309)

Then you will just need to change the stories, and `isBidirectional` getter for `ScrollBox` and cleanup the `bidirectionalScroll` setting and docs for it.

Also please add the `bidirectional` option to a types sections of [all ScrollBox stories](https://github.com/pixijs/ui/tree/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/stories/scrollBox) where there is this setting.

Other than that it looks good. Thank you for the contribution ;)

bbazukun123 commented:
> Please add a `bidirectional` option to a `export type ListType = 'horizontal' | 'vertical';` of [List.ts](https://github.com/pixijs/ui/blob/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/List.ts#L309)
> 
> This way it will be consistent and will not require additional setting for the `Scrollbox`, together with that it will improve the `type` setting of the `List` as now it is not cleat enough.
> 
> It will just work out of the box for `List` as this is essentially how it works when `type` is undefined. So you will just have to add
> 
> ```
> case 'bidirectional':
> default:
> ```
> 
> [over here](https://github.com/pixijs/ui/blob/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/List.ts#L309)
> 
> Then you will just need to change the stories, and `isBidirectional` getter for `ScrollBox` and cleanup the `bidirectionalScroll` setting and docs for it.
> 
> Also please add the `bidirectional` option to a types sections of [all ScrollBox stories](https://github.com/pixijs/ui/tree/b8a6e7b1e7e6b081b1f6a78488cc126c9b561fd3/src/stories/scrollBox) where there is this setting.
> 
> Other than that it looks good. Thank you for the contribution ;)

Thanks Dima! I have now addressed the feedback accordingly and also add the `bidirectional` option to the types for the List stories as well - Hope they are all okay now 👍

CyberDex reviewed (approved):
Great job!
